presto-hive/src/main/java/com/facebook/presto/hive/metastore/glue/GlueHiveMetastore.java

```diff
@@ -601,6 +611,25 @@ public void dropColumn(String databaseName, String tableName, String columnName)
     }
 
     private List<Partition> getPartitions(String databaseName, String tableName, String expression)
+    {
+        List<Partition> partitions;
+        if (numSegments <= 1) {
+            partitions = getPartitions(databaseName, tableName, expression, null);
+        }
+        else {
+            // Do parallel partition fetch
+            List<Segment> segments = IntStream.range(0, numSegments)
+                    .mapToObj(s -> new Segment().withSegmentNumber(s).withTotalSegments(numSegments))
+                    .collect(toList());
+            partitions = segments.parallelStream()
```

Comment:
i think parallel streams (forkjoinpool) is not safe approach to parallelism, since we do not control how many such threads will be running concurrently and we don't control the pool size.

@dain @electrum ?

Reply:
I thought about using an explicit thread pool instead of using the default ForkJoinPool, but seemed like an overkill since this is low concurrency and also not an in inner loop. 

Would you recommend creating a thread pool inside GlueHiveMetastore with number of threads == number of segments? Is there another shared thread pool we can make use of?

Reply:
We avoid parallel streams entirely in Presto. Using an `ExecutorService` is preferred since the semantics around concurrency, thread naming, cancellation on failure, etc., are better understood by everyone.

The default ForkJoinPool should not be used, since the concurrency is not configurable and it is shared by everything using parallel streams with the default pool. The latter is important since streams are designed for CPU intensive operations -- if you block on IO, it will affect anything else using the pool.

Reply:
@electrum we could try tweak the sys props queried by `java.util.concurrent.ForkJoinPool#makeCommonPool`. For example, by binding some throwing thread factory via `java.util.concurrent.ForkJoinPool.common.threadFactory` (eg in tests), we could prevent code from running in the default FJP. I am not entirely sure such a drastic catch-all measure is a good idea. What do you think?